lib/revision.js

```diff
@@ -57,10 +57,9 @@ function run() {
       if(typeof arr[0] === 'string'){
         return arr[0];
       }else{
-        return Buffer.concat(arr).toString();
```

Comment:
Sorry my bad. Didn't encounter any problems with this and it was quite late when I did it (previously I have rewritten `stream-to-array` to realise this was the problem 😄). Sure thing It would break it. I'll look into it in a sec.

And the issue is (as stated in PR title) I cannot open any jpgs or pngs that would undergo the revision.

Reply:
So now replacing in JS works (to be honest I thought that this package is actually lacking this feature). And to why it previously didn't work I suspect it is due to the fact that JPGs and PNGs are binary and toString doesn't handle it good.

Reply:
Hey,

The [`Soup`](https://github.com/hexojs/hexo-asset-pipeline/pull/30/files#diff-401afab61b7b622c9c6dfc93a75438f8L83) function also expects a string as input. So again it will break the execution.

What do you think?